presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -604,7 +609,7 @@ public void onFailure(Throwable failedReason)
                     doRemoveRemoteSource(errorTracker, request, future);
                 }
                 else {
-                    errorRateLimit.addListener(() -> doRemoveRemoteSource(errorTracker, request, future), errorScheduledExecutor);
+                    errorRateLimit.addListener(() -> doRemoveRemoteSource(errorTracker, request, future), taskEventLoop);
```

Comment:
Run the callback itself in the event loop as well

Reply:
Good point. Thanks for catching this!